engine/src/scheme.rs

```diff
@@ -1,29 +1,53 @@
 use crate::{
     ast::FilterAst,
     functions::Function,
-    lex::{complete, expect, span, take_while, LexErrorKind, LexResult, LexWith},
-    types::{GetType, Type},
+    lex::{complete, expect, skip_space, span, take_while, LexErrorKind, LexResult, LexWith},
+    types::{GetType, RhsValue, Type},
 };
 use failure::Fail;
 use fnv::FnvBuildHasher;
 use indexmap::map::{Entry, IndexMap};
-use serde::{Deserialize, Serialize, Serializer};
+use serde::{ser::SerializeSeq, Deserialize, Serialize, Serializer};
 use std::{
     cmp::{max, min},
     error::Error,
     fmt::{self, Debug, Display, Formatter},
+    iter::Iterator,
     ptr,
 };
 
-#[derive(PartialEq, Eq, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize)]
+#[serde(untagged)]
+pub enum FieldPathItem {
+    MapKey(String),
+}
+
+// special case for simply passing strings
+impl From<&str> for FieldPathItem {
```

Comment:
This should probably accept `String` in the first place instead of performing an implicit clone & allocation internally.

Reply:
Then we would have to do `"foo".to_string().into()` rather than `"foo".into()`?